bigframes/core/compile/compiled.py

```diff
@@ -214,7 +221,7 @@ def peek_sql(self, n: int):
         # Peek currently implemented as top level LIMIT op.
         # Execution engine handles limit pushdown.
         # In future, may push down limit/filters in compilation.
-        sql = ibis_bigquery.Backend().compile(self._to_ibis_expr().limit(n))
+        sql = ibis_backend.compile(self._to_ibis_expr().limit(n))
```

Comment:
I don't think compilation will need credentials, but if it does this is potentially out of sync with the credentials used in the Session.